src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -24,10 +24,50 @@ internal sealed class Deflater : IDisposable
         // Note, DeflateStream or the deflater do not try to be thread safe.
         // The lock is just used to make writing to unmanaged structures atomic to make sure
         // that they do not get inconsistent fields that may lead to an unmanaged memory violation.
-        // To prevent *managed* buffer corruption or other weird behaviour users need to synchronise
+        // To prevent *managed* buffer corruption or other weird behavior users need to synchronize
         // on the stream explicitly.
         private object SyncLock => this;
 
+        internal Deflater(ZLibCompressionOptions options, int windowBits)
+        {
+            Debug.Assert(windowBits >= minWindowBits && windowBits <= maxWindowBits);
+
+            int memLevel = options.CompressionLevel == 0 ? ZLibNative.Deflate_NoCompressionMemLevel : ZLibNative.Deflate_DefaultMemLevel;
```

Comment:
Has anyone requested being able to configure the memory level? I assume this is something we would consider adding as another property on the options in the future if it was needed?

Reply:
No real request, yes it can be added when needed